### PR TITLE
330 add srid to geometry types on write

### PR DESCRIPTION
**Pull request by adp-atea**

### CODE REVIEW

File: spinta/backends/postgresql/commands/write.py [hunk 1]
```diff
@@ -22,7 +22,11 @@ async def insert(
     table = backend.get_table(model)
     async for data in dstream:
         patch = commands.before_write(context, model, backend, data=data)
-
+        if 'coordinates' in patch.keys():
+            if 'SRID' not in patch['coordinates']:
+                coordinates = cast(sa.Column, backend.tables[model.name].columns['coordinates'])
+                srid = coordinates.type.srid
+                patch['coordinates'] = 'SRID='+str(srid)+';'+str(patch['coordinates'])
```
Comment (sirex) [inline spinta/backends/postgresql/commands/write.py:29]: This should not be gere, all geometry related code should go under geometry related command methods.

File: spinta/backends/postgresql/commands/write.py [hunk 2]
```diff
@@ -22,7 +22,11 @@ async def insert(
     table = backend.get_table(model)
     async for data in dstream:
         patch = commands.before_write(context, model, backend, data=data)
-
+        if 'coordinates' in patch.keys():
```
Comment (sirex) [inline spinta/backends/postgresql/commands/write.py:25]: What if a property is not named `coordinates`? See my testing notes, I named it `point`.

File: tests/dtypes/test_geometry.py [hunk 3]
```diff
@@ -150,6 +149,25 @@ def test_geometry_params_srid(
...
+    assert listdata(resp, full=True) == [
+        {
+            'name': "Vilnius",
+            'coordinates': 'POINT (582710 6061887)',
+        },
+        {
+            'name': "Vilnius",
+            'coordinates': 'POINT (582710 6061887)',
+        }
+    ]
```
Comment (sirex) [inline tests/dtypes/test_geometry.py:169]: Please, move it to another test, that tests case when SRID is given in geometry itself.

**Review by sirex — CHANGES_REQUESTED** [review]

File: spinta/backends/postgresql/commands/write.py [hunk 4]
```diff
@@ -133,3 +132,22 @@ def after_write(
...
+def before_write(
+    context: Context,
+    dtype: Geometry,
+    backend: PostgreSQL,
+    *,
+    data: DataSubItem):
+    patch = {}
+    if 'SRID' not in data.patch and dtype.srid is not None:
+        value = {dtype.prop.name: 'SRID='+str(dtype.srid)+';'+str(data.patch)}
+        patch.update(value)
```
Comment (sirex) [inline spinta/backends/postgresql/commands/write.py:147]:
```suggestion
        patch = {dtype.prop.name: 'SRID='+str(dtype.srid)+';'+str(data.patch)}
```

File: spinta/backends/postgresql/commands/write.py [hunk 5]
```diff
@@ -133,3 +132,22 @@ def after_write(
...
+    else:
+        value = {dtype.prop.name: str(data.patch)}
+        patch.update(value)
```
Comment (sirex) [inline spinta/backends/postgresql/commands/write.py:150]:
```suggestion
        patch = {dtype.prop.name: str(data.patch)}
```

**Review by sirex — CHANGES_REQUESTED** [review]

File: tests/dtypes/test_geometry.py [hunk 6]
```diff
@@ -287,3 +277,46 @@ def test_loading(tmp_path: Path, rc: RawConfig):
...
+):
+    context = bootstrap_manifest(rc, f'''
+        d | r | b | m | property                | type           | access
+        backends/postgres/dtypes/geometry/srid  |                | 
+          |   |   | Point                       |                | 
+          |   |   |   | point                   | geometry       | open
+          |   |   | PointLKS94                  |                | 
+          |   |   |   | point                   | geometry(3346) | open
+          |   |   | PointWGS84                  |                | 
+          |   |   |   | point                   | geometry(4326) | open
```
Comment (sirex) [inline tests/dtypes/test_geometry.py:295]: Why you need `Point` and `PointWGS84` if you only test with `PointLKS94`?

File: tests/dtypes/test_geometry.py [hunk 7]
```diff
@@ -287,3 +277,46 @@ def test_loading(tmp_path: Path, rc: RawConfig):
...
+
+    resp = app.post(f'/{model}', json={
+        'point': 'POINT(582710 6061887)',
+    })
+    assert resp.status_code == 201
+
+    resp = app.post(f'/{model}', json={
+        'point': 'POINT(582710 6061887)',
+    })
+    assert resp.status_code == 201
```
Comment (sirex) [inline tests/dtypes/test_geometry.py:320]: Why you need to add exactly same thing, two times?

**Review by sirex — CHANGES_REQUESTED** [review]

File: tests/dtypes/test_geometry.py (hunk 7 above)
Comment (adp-atea) [reply tests/dtypes/test_geometry.py:332]: Missed that.

**Review by sirex — APPROVED** [review]

**Pull request merged**